luigi/contrib/kubernetes.py

```diff
@@ -181,27 +203,90 @@ def signal_complete(self):
         """
         pass
 
+    def __get_pods(self):
+        pod_objs = Pod.objects(self.__kube_api) \
+            .filter(selector="job-name=" + self.uu_name) \
+            .response['items']
+        return [Pod(self.__kube_api, p) for p in pod_objs]
+
+    def __get_job(self):
+        jobs = Job.objects(self.__kube_api) \
+            .filter(selector="luigi_task_id=" + self.job_uuid) \
+            .response['items']
+        assert len(jobs) == 1, "Kubernetes job " + self.uu_name + " not found"
+        return Job(self.__kube_api, jobs[0])
+
+    def __print_pod_logs(self):
+        for pod in self.__get_pods():
+            logs = pod.logs(timestamps=True).strip()
+            self.__logger.info("Fetching logs from " + pod.name)
+            if len(logs) > 0:
+                for l in logs.split('\n'):
+                    self.__logger.info(l)
```

Comment:
This is very nice to have, but I am a bit concerned about messy mixed output from parallel jobs. We have some use cases in which we run 50 jobs in parallel, and it works nicely in terms of performance (with no need of Spark/Hadoop). What do you think?

Reply:
Interesting point - I guess we can change the logger to output the worker id so interleaved logs and be used easily. 
The best solution to me seems to be to do less logging within luigi (this code is turned off by default now) and more within the docker container, and then collect the container logs.

Reply:
Yep, good to have it optional for now.